src/include/network/lan/address.rb

```diff
@@ -1614,6 +1614,17 @@ def repeated_physical_port_ids(slaves)
 
     # Performs hostname update
     #
+    # This handles ip and hostname change when editing NIC properties.
+    # The method relies on old NIC's IP which is set globally at initialization
+    # of NIC edit dialog (@see LanItems#ipaddr)
+    #
+    # When hostname is empty, then old IP's record is cleared from /etc/hosts and
+    # new is not created.
+    # When hostname is changed then canonical name and all aliases in the record
+    # are replaced by new ones.
+    # When NIC's IP was changed than the canonical name and all aliases are assigned
+    # the new IP (or replaced by new ones if the hostname was changed as well)
```

Comment:
and it still clear old IP from /etc/hosts if I get code.

Reply:
yes, bcs it is change of ip in the particular configuration - no sense to keep it there.